src/Microsoft.VisualStudio.SDK.Analyzers.CodeFixes/build/AdditionalFiles/BannedSymbols.txt

```diff
@@ -1,2 +1,3 @@
 ﻿P:System.Net.ServicePointManager.SecurityProtocol; Modifying this would affect Visual Studio's process wide networking state.
 M:Microsoft.VisualStudio.Shell.Interop.IVsOutputWindowPane.OutputString(System.String); Prefer OutputWindowTextWriter for a free-threaded alternative. If that is not feasible, prefer IVsOutputWindowPaneNoPump.OutputStringNoPump (free-threaded) or OutputStringThreadSafe (not free-threaded).
+T:Microsoft.VisualStudio.Threading.AsyncSemaphore; Waiting for this semaphore on the UI thread can hang Visual Studio. Use the JTF-aware ReentrantSemaphore via ReentrantSemaphore.Create(initialCount, jtfContext, ReentrantSemaphore.ReentrancyMode.NotRecognized).
```

Comment:
Are you sure you want to recommend `NotRecognized`? That'll hang on reentrancy.

Reply:
That's implements the same behavior as AsyncSemaphore if I'm reading the docs correctly.

Reply:
I like the idea of focusing on the use rather than every reference to the type. As it is, this PR will make upgrading the analyzers in other repos (especially the VS repo) a longer process as we have a bunch of suppressions to add and possibly bugs to file.